### PR TITLE
Test packages building

### PR DESCRIPTION
**Pull request by ikeough**
### Purpose

- Adjusts the copy tasks on the Dynamo Samples libraries to copy assets into a package in the build folder. This allows us to maintain cross-platform copy functionality (AppData is locked on Linux). 
- The `CallsiteTests` are updated to load packages from this "testing" packages directory.
- Augments `DynamoModelTestBase` with two methods that allow custom setting of app data folders without having to override any of the test setup methods.

### Declarations

- [x] The code base is in a better state after this PR
  - `DynamoModelTestBase` now allows configuration of the data folders on the `IPathResolver` without needing to override the `StartDynamo` method.
  - The code is now safe for cross platform environments where `$(AppData)` can not be used.
  - The packages to be used for testing are isolated.
- [x] The level of testing this PR includes is appropriate.
  - This addition re-enables tests disabled by missing packages on the build machine.
### Reviewers

@pboyer

### CODE REVIEW

File: src/Libraries/Samples/SampleLibraryTests/SampleLibraryTests.csproj [hunk 1]
```diff
@@ -1,9 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
-  <ImportGroup Label="PropertySheets">
-    <Import Project="$(SolutionDir)Config/CS.props" />
-  </ImportGroup>
   <PropertyGroup>
```
Comment (pboyer) [inline src/Libraries/Samples/SampleLibraryTests/SampleLibraryTests.csproj]: Do we really want to remove this?

File: test/DynamoCoreTests/DynamoModelTestBase.cs [hunk 2]
```diff
@@ -91,6 +98,20 @@ protected virtual void GetLibrariesToPreload(List<string> libraries)
...
+            return string.Empty;
+        }
+
+        protected virtual string GetCommonDataRootFolder()
+        {
+            // Override in derived classed to provide a custom
+            // CommonAppDataRootFolder
+            return string.Empty;
+        }
+
```
Comment (pboyer) [inline test/DynamoCoreTests/DynamoModelTestBase.cs]: I'm assuming passing String.Empty will cause the default to be set in PathResolver.  Is that right?  We might want to clarify that point.

**Comment by pboyer** [comment]
@ikeough Just two minor points here.  However, let's make sure we're reverting my change here:

https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/Samples/SampleLibraryZeroTouch/SampleLibraryZeroTouch.csproj#L76  

I'm not sure why the Condition isn't present in the diff above...

File: src/Libraries/Samples/SampleLibraryTests/SampleLibraryTests.csproj (hunk 1 above)
Comment (ikeough) [inline src/Libraries/Samples/SampleLibraryTests/SampleLibraryTests.csproj]: Yes. We don't use it any more. We build these projects into their own \bin directories, so we don't need to take advantage of the variables set in the props file.

**Comment by pboyer** [comment]
Let's just wait for the green light and we're good.

**Pull request merged**